### PR TITLE
Add Contour 1.21.0 package

### LINKED ISSUE #4324: Contour package field to enable ipv6 (for Contour xDS server listeners/addresses and Envoy listeners/addresses)
## Feature Request
The Contour package should enable users to opt in to deploying Contour with ipv6 cluster compatible flags for listener addresses etc. This includes addresses for Contour and Envoy (e.g. flags like `--xds-address` and `--envoy-service-http-address` in the Contour Deployment). Ideally this would be a single value to configure, e.g. `contour.listenerIpFamily` or something similar.

## Describe alternatives you've considered
- Make users manually set flags for listen addresses etc.? Not really a good experience

## Additional context
The [cluster configuration](https://projectcontour.io/docs/v1.20.1/configuration/#cluster-configuration) has a field for lookup family used for ExternalName services, we may have to add a caveat to the documentation that if you're using an ipv6 only cluster, you should make sure to set this appropriately as well. It is a lone field in the embedded YAML of the values, so not sure if we can override it?

cc @vmware-tanzu/contour-owners @vmware-tanzu/pkg-contour-owners

### LINKED ISSUE #4605: add Contour v1.21.1 package
## Feature Request

Contour v1.21.1 has been released, we need to add packaging for it.

https://github.com/projectcontour/contour/releases/tag/v1.21.1

cc @vmware-tanzu/pkg-contour-owners

### PR DESCRIPTION
**Pull request by sunjayBhatia**
## What this PR does / why we need it
Adds Contour 1.21.0 package

Modifies change-namespace overlay to accommodate new RoleBinding and
ServiceAccount pulled out of ClusterRoleBinding for namespace specific
access to leader election resources. Also adds ipv6Enabled flag from
PR #4594

## Which issue(s) this PR fixes
Fixes: #4605
Fixes: #4324

## Describe testing done for PR
Unit testing and deployed interpolated YAML in kind cluster.

## Special notes for your reviewer
N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Also looking into https://github.com/vmware-tanzu/community-edition/issues/4605#issuecomment-1137462249 and how it interacts with downstream users of the package

File: addons/packages/contour/1.21.0/test/unittest/contour_test.go [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+			Expect(err).NotTo(HaveOccurred())
+
+			// Have to be fiddly with the whitespace here to avoid matching
+			// lines in the config file example and in the ContourConfiguration CRD
+			// schema.
+
+			docs := findDocsContainingLines(output, "\n  namespace: projectcontour")
+			Expect(docs).To(BeEmpty())
+
+			// Note, the overlay superficially indicates 16 matches (13 resources in the
```
Comment (sunjayBhatia) [inline addons/packages/contour/1.21.0/test/unittest/contour_test.go:385]: note the change from previous version here

File: addons/packages/contour/1.21.0/bundle/config/overlays/change-namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+
+#@overlay/match by=overlay.subset({"kind":"Namespace", "metadata": {"name": "projectcontour"}}), expects=1
+---
+metadata:
+  name: #@ data.values.namespace
+
+#! ignore cert-manager objects, since the code for generating them is already setting the namespace correctly.
+#@overlay/match by=overlay.and_op(overlay.subset({"metadata": {"namespace": "projectcontour"}}), overlay.not_op(overlay.subset({"apiVersion": "cert-manager.io/v1"}))), expects=13
```
Comment (sunjayBhatia) [inline addons/packages/contour/1.21.0/bundle/config/overlays/change-namespace.yaml:10]: note the change in number of expected resources here

File: addons/packages/contour/1.21.0/bundle/config/overlays/change-namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+---
+metadata:
+  namespace: #@ data.values.namespace
+
+#@overlay/match by=overlay.or_op(overlay.subset({"kind":"ClusterRoleBinding"}), overlay.subset({"kind":"RoleBinding"})), expects=3
```
Comment (sunjayBhatia) [inline addons/packages/contour/1.21.0/bundle/config/overlays/change-namespace.yaml:15]: and here

File: addons/packages/contour/1.21.0/package.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  version: 1.21.0
+  releaseNotes: "contour 1.21.0 https://github.com/projectcontour/contour/releases/tag/v1.21.0"
+  releasedAt: 2022-02-24T00:00:00Z
+  licenses:
+    - "Apache 2.0"
+  template:
+    spec:
+      fetch:
+        - imgpkgBundle:
+            image: projects.registry.vmware.com/tce/contour@sha256:870ad5b8ef974a0ed78126a2eb0a41e1a8a11413352667a1fd20afe07b0a1ebb
```
Comment (sunjayBhatia) [inline addons/packages/contour/1.21.0/package.yaml:16]: used image tag `1.21.0_build.1`, not sure if we should document somewhere (besides our internal instructions)

File: addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+images:
+- annotations:
+    kbld.carvel.dev/id: docker.io/envoyproxy/envoy:v1.22.0
+    kbld.carvel.dev/origins: |
+      - resolved:
+          tag: v1.22.0
+          url: docker.io/envoyproxy/envoy:v1.22.0
+      - preresolved:
+          url: index.docker.io/envoyproxy/envoy@sha256:e9ad4cde73ffb01402c4e3fb6ce1a764d9e84650230135b88495f30904a378fc
+  image: index.docker.io/envoyproxy/envoy@sha256:e9ad4cde73ffb01402c4e3fb6ce1a764d9e84650230135b88495f30904a378fc
```
Comment (skriss) [inline addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml:12]: This doesn't look to have changed from the 1.20.1 package, the `make lock-package-images` should update this but if it's not, need to dig into what's going wrong

File: addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+- annotations:
+    kbld.carvel.dev/id: ghcr.io/projectcontour/contour:v1.21.0
+    kbld.carvel.dev/origins: |
+      - resolved:
+          tag: v1.21.0
+          url: ghcr.io/projectcontour/contour:v1.21.0
+      - preresolved:
+          url: ghcr.io/projectcontour/contour@sha256:6e6c6cbe03eba0df8987da4cac2712845f66c496185383d8768f19b40be26125
+  image: ghcr.io/projectcontour/contour@sha256:6e6c6cbe03eba0df8987da4cac2712845f66c496185383d8768f19b40be26125
```
Comment (skriss) [inline addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml:21]: This doesn't look to have changed from the 1.20.1 package, the `make lock-package-images` should update this but if it's not, need to dig into what's going wrong

**Review by skriss — CHANGES_REQUESTED** [review]
Looks pretty good, just one comment on the kbld output

File: addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml (hunk 5 above)
Comment (sunjayBhatia) [reply addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml:12]: hm, yeah that command only added that `preresolved` section and I didn't really think about it, I'll look into it

Comment (skriss) [reply addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml:12]: Yeah I feel like maybe i've run into this before and probably failed to document - something with docker auth I think?

Comment (sunjayBhatia) [reply addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml:12]: hm, i should be logged in and nothing i seem to do, short of change the image sha is changing anything, `make lock-package-images` keeps just adding a new `preresolved` list element

Comment (sunjayBhatia) [reply addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml]: updated manually for now, will ask if theres something we need to do here

File: addons/packages/contour/1.21.0/package.yaml (hunk 4 above)
Comment (sunjayBhatia) [reply addons/packages/contour/1.21.0/package.yaml]: now pushed with tag 1.21.0_build.2

File: addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml (hunk 5 above)
Comment (sunjayBhatia) [reply addons/packages/contour/1.21.0/bundle/.imgpkg/images.yml]: ah, i think i figured it out, I had done a find-replace on the whole 1.21.0 package and that included the imgpkg lock file, when I reverted it, the images were resolved correctly

maybe something to bring up in kbld, not to be so trusting even if you find the lock file has a matched image?

**Pull request closed without merging**